### DISCUSSION TITLE
Does DSG use the source server for data/how can I prevent a source plugin from retrieving ALL data during a clean build?

### DISCUSSION THREAD

**Discussion by tsdexter** [gatsbyjs/gatsby#35459] [Help]
I would like to use DSG to speed up build times however I'm confused about exactly how it works. When the page is being built with DSG does the build server go back to my WordPress GraphQL endpoint and fetch the data or does it just get it from the cached Gatsby data layer?

In other words, if I use DSG for every page, is `gatsby-source-wordpress` still going to call my graphql endpoint for all of the unnecessary site data during a full build? This step alone is often timing out my builds even on an enterprise wp engine server. I would ideally like to have something set up where even a full, clean build does not need to source every piece of data from wordpress...is this possible?

**Comment by cavemon** [comment]
@tsdexter I don't think DSG makes a subsequent call to your server.  If I understand correctly, the Gatsby data layer that is generated during build is essentially kept live on the build server.  This data layer is what you typically interact with when using `gatsby develop` and visiting `http://localhost:8000/__graphql`.  So when a user requests the page and the cache miss occurs, it pulls from this data layer, but doesn't consist of realtime data if the data on your content source changed.  

I, too, am working on optimizing build times and since `gatsby-source-wordpress` takes so long to gather the data to build the data layer, I'm not sure how much time will be saved by leveraging DSG, but we'll see 🤷

**Reply by tsdexter** [reply]
That’s correct. DSG won’t/didn’t speed up build times for us (barely anyway). The build time benefit of DSG is just savings on the HTML rendering step, which is only a few seconds/< 1 min for our ~5k pages. We choose to keep that step on the build server, rather than using extra resources/time for initial page loads on the web server. 

We aren’t sourcing any images from wordpress, just page/post/custom post type data. We still serve the images from the wordpress server which has a CDN and is just as fast for static images. Our clean build times run around 17-28 mins. Cached builds are less than 10, iirc. With images we would always timeout at 30 min (previous, now 57 min) build limit on our WP Engine Atlas hosting
